Source/WebCore/rendering/RenderLayerModelObject.cpp

```diff
@@ -84,7 +85,8 @@ void RenderLayerModelObject::willBeDestroyed()
 
 void RenderLayerModelObject::willBeRemovedFromTree(IsInternalMove isInternalMove)
 {
-    if (auto* layer = this->layer(); layer && layer->needsFullRepaint() && isInternalMove == IsInternalMove::No)
+    bool shouldNotRepaint = this->previousSibling() && is<RenderMultiColumnSet>(this->previousSibling());
```

Comment:
is<> can take null_ptr and I'd not put this in-between the layer checks, but maybe as a trailing condition